tests/pipelines/test_pipelines_common.py

```diff
@@ -1687,7 +1687,15 @@ def _test_xformers_attention_forwardGenerator_pass(
             self.assertLess(max_diff, expected_max_diff, "XFormers attention should not affect the inference results")
 
         if test_mean_pixel_difference:
-            assert_mean_pixel_difference(output_with_offload[0], output_without_offload[0])
+            if torch.is_tensor(output_without_offload):
```

Comment:
If the issue is the output shapes it would be better to redefine the tests for that output shape inside the pipeline test modules. Similar to what's done here:
https://github.com/huggingface/diffusers/blob/9ab80a99a4f57b61dc1c4027d7e0fcb13a602aad/tests/pipelines/animatediff/test_animatediff.py#L467